src/client/dfuse/ops/rename.c

```diff
@@ -29,6 +32,18 @@ dfuse_cb_rename(fuse_req_t req, struct dfuse_inode_entry *parent,
 		       name, newname, newparent);
 
 	DFUSE_REPLY_ZERO(parent, req);
+
+	if (parent->ie_dfs->dfc_dentry_timeout ||
+		parent->ie_dfs->dfc_dentry_dir_timeout ||
+		parent->ie_dfs->dfc_ndentry_timeout) {
+		/* The caller has a reference on newparent */
+		rc = fuse_lowlevel_notify_inval_entry(fs_handle->dpi_info->di_session,
+						      newparent->ie_stat.st_ino,
+						      newname, strnlen(newname, NAME_MAX));
+		if (rc)
+			DFUSE_TRA_ERROR(parent,
+					"inval_entry failed %d", rc);
+	}
```

Comment:
I'm unhappy with this as it seems there's a race condition where the kernel or other processes might end up doing a lookup which would result in ENOENT before we've had a chance to invalidate the dentry.  I'm working on a standalone reproducer which I'll send to the fuse maintainers to seek their opinions.